test/jdk/javax/swing/JFileChooser/bug4522756.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2002, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4522756
+ * @requires (os.family == "windows")
+ * @summary To verify that if for the first time JFileChooser is opened,
+            the icon for Desktop is not missing.
```

Comment:
```suggestion
 * @summary Verifies that the Desktop icon is not missing when
            JFileChooser is opened for the first time.
```

I presume it refers to the location bar on the left, does it?

Reply:
The original bug was on GTK+linux only. I have extended it to SystemLookAndFeel. Since there are slight variations on how the file filter works on different LaF and the test instructions would be confusing if we were to account for all variations of installed LaF. Hence limited it System LaF. For example on macOS (Aqua LaF) when the txt filter is applied non-text files are greyed out.